app/(main)/leaderboard/agents/AgentsLeaderboardTable.tsx

```diff
@@ -47,18 +48,31 @@ export default function AgentsLeaderboardTable() {
   const initialSortKey = (searchParams.get('sort') as SortKeyType) || defaultSort;
   const initialSortOrder =
     (searchParams.get('order')?.toLowerCase() as SortOrder) || defaultOrder;
+  const initialFilter = searchParams.get('filter') || '';
 
   const [page, setPage] = useState(initialPage);
   const [sortKey, setSortKey] = useState<SortKeyType>(initialSortKey);
   const [sortOrder, setSortOrder] = useState<SortOrder>(initialSortOrder);
+  const [filter, setFilter] = useState(initialFilter);
+  const [inputValue, setInputValue] = useState(initialFilter);
+
+  const debouncedFilter = useDebounce(inputValue, 900);
 
   const { data, isLoading } = useQuery({
-    queryKey: ['getAgentsTradeMetricsData', page, ITEMS_PER_PAGE, sortKey, sortOrder],
+    queryKey: [
+      'getAgentsTradeMetricsData',
+      page,
+      ITEMS_PER_PAGE,
+      sortKey,
+      sortOrder,
+      filter,
+    ],
     queryFn: () =>
       getAgentsTradeMetricsData({
         page,
         pageSize: ITEMS_PER_PAGE,
         sort_by: `${sortKey} ${sortOrder}`,
+        filters: filter ? `address = ${filter} or label = ${filter}` : '',
```

Comment:
Is there some sort of `like` or `like_operator` we could use here? Right now it works for exact matches (at least what I could test so far) but it would be awesome searching for "olas" and getting all the names that contains `olas` (upper case, lower case, etc)

Reply:
yes, it would be awesome, I'm trying to see how can we do that since it's not clear on Dune docs, but I'm their discord trying to figure it out.

Reply:
done!